api/topo/topo.go

```diff
@@ -33,3 +58,86 @@ var TopoClientFactory = func(cc *grpc.ClientConn) TopoClient {
 func CreateTopoClient(cc *grpc.ClientConn) TopoClient {
 	return TopoClientFactory(cc)
 }
+
+// ObjectToDevice ...
+func ObjectToDevice(obj *Object) *device.Device {
+	if obj == nil || obj.Type != Object_ENTITY { // Device is an entity
+		return nil
+	}
+
+	d := &device.Device{
+		ID:        device.ID(obj.ID),
+		Protocols: []*device.ProtocolState{},
+	}
+
+	r, _ := strconv.Atoi(obj.Attributes[Revision])
+	d.Revision = device.Revision(r)
+	d.Address = obj.Attributes[Address]
+	d.Target = obj.Attributes[Target]
+	d.Version = obj.Attributes[Version]
+	t, _ := strconv.Atoi(obj.Attributes[Timeout])
+	timeout := time.Duration(t) * time.Second
+	d.Timeout = &timeout
+	d.Role = device.Role(obj.Attributes[Role])
+	d.Displayname = obj.Attributes[Displayname]
+	d.Credentials.User = obj.Attributes[User]
+	d.Credentials.Password = obj.Attributes[Password]
+	d.TLS.CaCert = obj.Attributes[TLSCaCert]
+	d.TLS.Cert = obj.Attributes[TLSCert]
+	d.TLS.Key = obj.Attributes[TLSKey]
+	if strings.ToLower(obj.Attributes[TLSPlain]) == "true" {
+		d.TLS.Plain = true
+	} else if strings.ToLower(obj.Attributes[TLSPlain]) == "false" {
+		d.TLS.Plain = false
+	}
+	if strings.ToLower(obj.Attributes[TLSInsecure]) == "true" {
+		d.TLS.Insecure = true
+	} else if strings.ToLower(obj.Attributes[TLSInsecure]) == "false" {
+		d.TLS.Insecure = false
+	}
+	entity := obj.GetEntity()
+	if entity != nil {
+		d.Protocols = obj.GetEntity().Protocols
+	}
+
+	return d
+}
+
+// DeviceToObject ...
+func DeviceToObject(d *device.Device) *Object {
+	obj := &Object{
+		ID:         ID(d.ID),
+		Type:       Object_ENTITY,
+		Attributes: map[string]string{},
+		Obj: &Object_Entity{
+			Entity: &Entity{
+				Protocols: []*device.ProtocolState{},
+			},
+		},
+	}
+	obj.Attributes[Revision] = strconv.Itoa(int(d.Revision))
+	obj.Attributes[Address] = d.Address
+	obj.Attributes[Target] = d.Target
+	obj.Attributes[Version] = d.Version
```

Comment:
missing `Type` - maybe it should be the `kind`

Reply:
done